hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/RecordLevelIndexSupport.scala

```diff
@@ -123,28 +123,47 @@ class RecordLevelIndexSupport(spark: SparkSession,
    * @param queryFilters The queries that need to be filtered.
    * @return Tuple of List of filtered queries and list of record key literals that need to be matched
    */
-  private def filterQueryFiltersWithRecordKey(queryFilters: Seq[Expression]): (List[Expression], List[String]) = {
+  private def filterQueriesWithRecordKey(queryFilters: Seq[Expression]): (List[Expression], List[String]) = {
     var recordKeyQueries: List[Expression] = List.empty
     var recordKeys: List[String] = List.empty
     for (query <- queryFilters) {
-      query match {
-        case equalToQuery: EqualTo =>
-          val (attribute, literal) = getAttributeLiteralTuple(equalToQuery.left, equalToQuery.right).orNull
-          if (attribute != null && attribute.name != null && attributeMatchesRecordKey(attribute.name)) {
-            recordKeys = recordKeys :+ literal.value.toString
-            recordKeyQueries = recordKeyQueries :+ equalToQuery
-          }
-        case _ =>
-      }
+      filterQueryWithRecordKey(query).foreach({
+        case (exp: Expression, recKeys: List[String]) =>
+          recordKeys = recordKeys ++ recKeys
+          recordKeyQueries = recordKeyQueries :+ exp
+      })
     }
+
     Tuple2.apply(recordKeyQueries, recordKeys)
   }
 
-  /**
+  private def filterQueryWithRecordKey(queryFilter: Expression): Option[(Expression, List[String])] = {
```

Comment:
can we add java docs on what are the supported filtering as of now.

Reply:
Addressed